ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/OfferingToStateMapper.kt

```diff
@@ -8,23 +8,21 @@ import com.revenuecat.purchases.ui.revenuecatui.data.processed.VariableDataProvi
 import java.util.Locale
 
 @Suppress("ReturnCount", "TooGenericExceptionCaught")
-internal fun Offering.toPaywallViewState(variableDataProvider: VariableDataProvider): PaywallViewState {
+internal fun Offering.toPaywallViewState(
+    variableDataProvider: VariableDataProvider,
+): PaywallViewState {
     val paywallData = this.paywall
         ?: return PaywallViewState.Error("No paywall data for offering: $identifier")
-    return try {
-        val templateConfiguration = TemplateConfigurationFactory.create(
-            variableDataProvider = variableDataProvider,
-            mode = PaywallViewMode.FULL_SCREEN,
-            paywallData = paywallData,
-            packages = availablePackages,
-            activelySubscribedProductIdentifiers = emptySet(), // TODO-PAYWALLS: Check for active subscriptions
-            locale = Locale.getDefault(), // TODO-PAYWALLS: Use the correct locale
-        )
-        PaywallViewState.Loaded(
-            templateConfiguration = templateConfiguration,
-            selectedPackage = templateConfiguration.packages.default,
-        )
-    } catch (e: Exception) {
-        PaywallViewState.Error("Error creating paywall: ${e.message}")
-    }
+    val templateConfiguration = TemplateConfigurationFactory.create(
```

Comment:
Looks like this method is still calling `PaywallTemplate.fromId(paywallData.templateName)` and failing if not found.
To avoid that I made the validation result produce the processed `PaywallTemplate`, so then we could inject it here. What do you think?

Reply:
Note that in iOS the `ValidationResult` also has the `displayablePaywall: PaywallData` which is how we can reuse the configured `packages` even if there are validation errors.

Reply:
Addressed 👍🏻